examples/createAndInitPool/createAndInitPoolV3.ts

```diff
@@ -29,112 +32,142 @@ import {
-        name: 'USDT stataDAI partially boosted',
-        symbol: 'USDT-stataDAI',
-        poolType: PoolType.Stable,
-        tokens: [
-            {
-                address: AAVE_FAUCET_USDT.address,
-                rateProvider: zeroAddress,
-                tokenType: TokenType.STANDARD,
-                paysYieldFees: false,
-            },
-            {
-                address: STATA_ETH_DAI.address,
-                rateProvider: STATA_ETH_DAI.rateProvider,
-                tokenType: TokenType.TOKEN_WITH_RATE,
-                paysYieldFees: true,
-            },
-        ],
-        amplificationParameter: BigInt(33),
-        swapFeePercentage: parseUnits('0.001', SWAP_FEE_PERCENTAGE_DECIMALS),
-        pauseManager: zeroAddress,
-        swapFeeManager: zeroAddress,
-        poolHooksContract: zeroAddress,
-        enableDonation: false,
-        disableUnbalancedLiquidity: false,
-        protocolVersion: 3,
-        chainId,
-    };
-    const initAmounts = [
-        {
-            address: AAVE_FAUCET_USDT.address,
-            rawAmount: parseUnits('10', AAVE_FAUCET_USDT.decimals),
-            decimals: AAVE_FAUCET_USDT.decimals,
-        },
+    // Approve the cannonical permit2 contract to spend tokens that will be used for pool init
+    await approveSpenderOnTokens(
+        client,
+        account,
+        amountsIn.map((t) => t.address),
+        PERMIT2[chainId],
+    );
+
+    // Build the init pool call
+    const initPoolCall = await buildInitPoolCall({
+        client,
+        rpcUrl,
+        account,
+        poolAddress,
+    });
+
+    // Send the init pool tx to local anvil fork
+    await makeForkTx(
+        initPoolCall,
         {
-            address: STATA_ETH_DAI.address,
-            rawAmount: parseUnits('10', STATA_ETH_DAI.decimals),
-            decimals: STATA_ETH_DAI.decimals,
+            rpcUrl,
+            chainId,
+            impersonateAccount: account,
+            forkTokens: amountsIn.map((a) => ({
+                address: a.address,
+                slot: getSlot(chainId, a.address),
+                rawBalance: a.rawAmount,
+            })),
         },
-    ];
+        [...amountsIn.map((a) => a.address), poolAddress],
+        createPoolInput.protocolVersion,
+    );
+}
 
-    // Build the create pool call using the SDK
-    const call = await createPoolCall(createPoolInput);
+export async function runAgainstNetwork() {
+    const rpcUrl = process.env.SEPOLIA_RPC_URL;
+    if (!rpcUrl) throw new Error('rpcUrl is undefined');
 
-    // Submit the create pool tx and wait for the PoolCreated event to find pool address
-    const poolAddress = await createPool({ client, call, userAccount });
-    console.log('Created Pool Address: ', poolAddress);
+    const client = createWalletClient({
+        chain: CHAINS[chainId],
+        transport: http(rpcUrl),
+        account: privateKeyToAccount(process.env.PRIVATE_KEY as `0x${string}`),
+    })
+        .extend(walletActions)
```

Comment:
I don't think this is required as its already a wallet client?

Reply:
you are correct ser!